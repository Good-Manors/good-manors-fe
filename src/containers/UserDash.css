.UserDash {
  text-align: center;
}

button {
  font-family: 'Sulphur', sans-serif;
  font-weight: 700;
  font-size: 1rem;
  letter-spacing: .08em;
  border: none;
  padding: .3rem 1rem;
  margin: 20px 0;
  background: #8ab4b1;
  color: white;
}

.buttonDefault {
  background: #247BA0;
  color: white;
}
