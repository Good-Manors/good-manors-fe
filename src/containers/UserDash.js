import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import Modal from '../components/modal/Modal';
import Header from '../components/Header/Header';
import Menu from '../components/Menu';
import useModal from '../hooks/useModal';
import styles from './UserDash.css';
import { getAllHomes, getHome } from '../services/homes';
import { setHome } from '../actions/homeActions';
import icon from '../assets/icons/custom-drawer-icon.png';

const UserDash = () => {
  const { isShowing, toggle } = useModal();
  const [userHomes, setUserHomes] = useState();
  const [redirect, setRedirect] = useState(false);

  const dispatch = useDispatch();

  useEffect(() => {
    getAllHomes()
      .then(homes => {
        if(!homes) {
          setUserHomes('none');
        }
        setUserHomes(homes);
      });
  }, []);

  const handleHomeClick = (id) => {
    getHome(id)
      .then(home => {
        dispatch(setHome(home));
        setRedirect(true);
      });
  };

  if(redirect) return <Redirect to='/home' />;

  if(!userHomes) return <h1>Loading</h1>;

  const mappedHomes = userHomes.map(home => {
    return <div className={styles.homeDisplay} key={home._id} onClick={() => handleHomeClick(home._id)}>
      <img src={icon} />
      <p>{home.title}</p>
    </div>;
  });

  if(userHomes === 'none') toggle();

  return (
    <section className={styles.UserDash}>
      <Header />
      <Menu />
      <section className={styles.UserDashWrapper}>
        <h2>Dashboard</h2>
        <button onClick={toggle}>+ New Home</button>
        <section className={styles.homeSection}>
          {mappedHomes}
        </section>
        <Modal
          isShowing={isShowing}
          hide={toggle}
        />
      </section>
    </section>
  );
};

export default UserDash;
