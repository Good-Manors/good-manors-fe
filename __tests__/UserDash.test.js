import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import UserDash from '../src/containers/UserDash';

describe('Renders a UserDash container', () => {
  it('Should render a UserDash', () => {
    const getWrapper = (mockStore = createStore()) => mount(
      <Provider store={mockStore}>
        <UserDash/>
      </Provider>
    );
    expect(getWrapper).toMatchSnapshot();
  });
});
