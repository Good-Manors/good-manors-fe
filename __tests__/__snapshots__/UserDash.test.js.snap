// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a UserDash container Should render a UserDash 1`] = `[Function]`;
